src/app/pages/storage/volumes/encyption-options-dialog/encryption-options-dialog.component.ts

```diff
@@ -0,0 +1,188 @@
+  }
+
+  onSubmit(): void {
+    if (this.form.value.inherit_encryption) {
+      // Only try to change to inherit if not currently inheriting
+      if (!this.data.row.is_encrypted_root) {
+        this.dialogRef.close(false);
+        return;
+      }
+
+      this.setToInherit();
+    } else {
+      this.saveForm();
+    }
+  }
+
+  private setToInherit(): void {
+    this.loader.open();
+    this.ws.call('pool.dataset.inherit_parent_encryption_properties', [this.data.row.id])
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.loader.close();
+        this.showSuccessDialog();
+        this.dialogRef.close(true);
+      }, (error: WebsocketError) => {
+        this.loader.close();
+        this.errorHandler.handleWsFormError(error, this.form);
+      });
+  }
+
+  private saveForm(): void {
+    const values = this.form.getRawValue();
+
+    const body = {} as DatasetChangeKeyParams;
+    if (values.encryption_type === EncryptionType.Key) {
+      body.generate_key = values.generate_key;
+      if (!values.generate_key) {
+        body.key = values.key;
+      }
+    } else {
+      body.passphrase = values.passphrase;
+      body.pbkdf2iters = Number(values.pbkdf2iters);
+    }
+
+    this.loader.open();
+    this.ws.call('pool.dataset.change_key', [this.data.row.id, body])
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.loader.close();
+        this.showSuccessDialog();
+        this.dialogRef.close(true);
+      }, (error: WebsocketError) => {
+        this.loader.close();
+        this.errorHandler.handleWsFormError(error, this.form);
+      });
+  }
+
+  private showSuccessDialog(): void {
+    this.dialog.info(
+      this.translate.instant('Encryption Options Saved'),
+      this.translate.instant('Encryption options for {id} successfully saved.', { id: this.data.row.id }),
+      '500px',
+      'info',
+    );
+  }
+
+  private setFormValues(): void {
+    let encryptionType = EncryptionType.Passphrase;
+    if (!this.data.hasPassphraseParent) {
+      if (this.data.hasKeyChild) {
+        encryptionType = EncryptionType.Key;
+      } else {
+        encryptionType = this.data.row.is_passphrase ? EncryptionType.Passphrase : EncryptionType.Key;
+      }
+    }
+
+    this.form.patchValue({
+      inherit_encryption: !this.data.row.is_encrypted_root,
+      algorithm: this.data.row.encryption_algorithm?.value || '',
+      encryption_type: encryptionType,
+    });
+
+    if (this.data.row.pbkdf2iters && this.data.row.pbkdf2iters.rawvalue !== '0') {
+      this.form.patchValue({
+        pbkdf2iters: Number(this.data.row.pbkdf2iters.rawvalue),
+      });
+    }
+  }
+
+  private setControlDependencies(): void {
+    this.form.controls.algorithm.disable();
+
+    if (this.data.hasPassphraseParent || this.data.hasKeyChild) {
+      this.form.controls['encryption_type'].disable();
+    }
+
+    this.form.controls['key'].disabledWhile(combineLatestIsAny([
```

Comment:
I noticed that `control.disableWhile()` internally calls `.subscribe` , but never calls `.unsubsribe` .  
Additionally, I found out that `form.select()` call won't stop emitting after component is destroyed.
I also noticed that `.disableWhile` returns a `Subscription` object, which brings to us the responsibility to call `.unsubscribe`.

Summarizing these points, I suggest to call `.unsubscribe` at `ngOnDestroy()`, or alternatively, to apply `untilDestroyed` pipe after `form.select()`.

Reply:
You are right 😔 
I'll fix it as a separate PR, because this is not the only place with `disabledWhile` and I want to discuss this with library authors.